File: wordle.py
# import random
# from colorama import Back, Style

# allowedWords=open("./wordle-allowed-guesses.txt")
# answersWords=open("./wordle-answers-alphabetical.txt")
# answersList=[n.rstrip() for n in answersWords.readlines()]
# allowedList=[n.rstrip() for n in allowedWords.readlines()]
# word=random.choice(answersList).rstrip()
# triesLeft=6
# found=False
# print(word)
# while True:
#     if triesLeft==0:
#         print("OUT OF TRIES")
#         print(Back.GREEN+word)
#         break
#     print(str(triesLeft)+" tries left.")
#     attempt=input().lower()
#     allowed=False
#     if len(attempt)==5:
#         if word in allowedList:
#             allowed=True
#     output=["N"]*5
#     foundLetters=[]
#     if allowed:
#         wordList=[n for n in word]
#         attemptList=[n for n in attempt]
#         for k in range(0,5):
#             if wordList[k]==attemptList[k]:
#                 output[k]="Y"
#                 foundLetters.append(attemptList[k])
#         for k in range(0,5):
#             if attemptList[k] in wordList:
#                 if attemptList[k] not in foundLetters:
#                     if attemptList[k] != "Y":
#                         output[k]="C"
#         if output==["Y"]*5:
#             found=True
#         string=""
#         for n,k in enumerate(output):
#             if k=="N":
#                 string=string+str(Back.RED+attempt[n])
#             elif k=="Y":
#                 string=string+str(Back.GREEN+attempt[n])
#             elif k=="C":
#                 string=string+str(Back.YELLOW+attempt[n])
#         print(string+Style.RESET_ALL)
#         triesLeft-=1
            
#     else:
#         print("INVALID INPUT")

#     if found:
#         break

 

File: wordle.js
function print(string){
  console.log(string)
}
const fs = require("fs");

answerList=[]
allowedList=[]
var inputs=["texts","midge","break","hello","lynch","adieu"]
var found=false
var word
var triesLeft=6
var input

async function getFile(file){
  return await fs.promises.readFile(file, "utf-8")
}

async function getWords(){

  data = await getFile("./wordle-allowed-guesses.txt")
  allowedList=data.split(/\r\n|\n\r|\n|\r/)
  data =  await getFile("./wordle-answers-alphabetical.txt")
  answerList=data.split(/\r\n|\n\r|\n|\r/)
  word= answerList[Math.floor(Math.random()*answerList.length)];
}

function inputCheck() {
  if (input.length==5){
    if (allowedList.includes(input)){
      allowed=true
    }
  }
}

function inputProcessing(){
  inputList=input.split("")
  print(inputList)
  wordList=word.split("")
  print(wordList)
  triesLeft=triesLeft-1
  for (let i = 0; i < 5; i++) {
    if (inputList[i]==wordList[i]){
      output[i]="Y"
    }
  }
  for (let i = 0; i < 5; i++) {
    if (wordList.includes(inputList[i])){
      if (foundLetters.includes(inputList[i])==false){
        if (output[i]!="Y"){
          output[i]="C"
        }
      }
    }
  }
}

async function main(){
  await getWords()
  
  while (triesLeft>0 && found==false){
    input=inputs[triesLeft-1]
    output=["N","N","N","N","N"]
    foundLetters=[]
    allowed=false
    inputCheck()
    if (allowed){
      print(triesLeft)
      inputProcessing()
    }else{
      print("INVALID IMPUT")
    }
    print(output)
  }
}

main()
